From 996891e061b2591a7b631af1a15b3a6d5f92ce24 Mon Sep 17 00:00:00 2001
From: Dongxu Sun <sundongxu3@huawei.com>
Date: Thu, 11 Jan 2024 20:43:20 +0800
Subject: [PATCH 544/960] downstream: mbigen: vtimer: disable vtimer mbigen
 probe when vtimer_irqbypass disabled

Hardware firmware may have different implement, disable vtimer mbigen
probe when vtimer_irqbypass disabled.

Signed-off-by: Dongxu Sun <sundongxu3@huawei.com>
Signed-off-by: Slim6882 <15605196882@163.com>
Signed-off-by: chenyi <chenyi211@h-partners.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 drivers/irqchip/irq-mbigen.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/irqchip/irq-mbigen.c b/drivers/irqchip/irq-mbigen.c
index 65f0fd350185..edc077c88d78 100644
--- a/drivers/irqchip/irq-mbigen.c
+++ b/drivers/irqchip/irq-mbigen.c
@@ -763,6 +763,9 @@ static int vtimer_mbigen_device_probe(struct platform_device *pdev)
 	struct vtimer_mbigen_device *vtimer_mgn_chip;
 	int err;
 
+	if (!vtimer_irqbypass)
+		return 0;
+
 	err = vtimer_mbigen_set_regs(pdev);
 	if (err)
 		return err;
-- 
2.43.0

